crates/recursor/src/error.rs

```diff
@@ -142,6 +143,29 @@ impl Error {
         )
         .into())
     }
+
+    /// Return an error if more than max CNAME records have been resolved in a CNAME chain.
+    pub fn cname_exceeded(
+        cname_limit: Arc<Mutex<u8>>,
```

Comment:
Do we need to expose the `Mutex` (particularly from a dependency) in the public API? Why?

Reply:
This was only called from one location (and I don't think that will change) so I inlined it.